File: linear_algebra/Python/BTAT_BT6.py
m, n = map(int, input("Nhap so hang va so cot cua ma tran A: ").split())
A = []
print("Nhap cac phan tu cua ma tran A:")
for i in range(m):
    A.append(list(map(int, input().split())))

p = int(input("Nhap kich thuoc vector b: "))
b = list(map(int, input("Nhap cac phan tu cua b: ").split()))

if m != p:
    print("Impossible")
else:
    print("Ma tran bo sung [A|b] la:")
    for i in range(m):
        for j in range(n):
            print(A[i][j], end=" ")
        print(b[i])

File: linear_algebra/C++/BTAT_BT6.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int m, n;
    cout << "Nhap so hang va so cot cua ma tran A: ";
    cin >> m >> n;

    vector<vector<int>> A(m, vector<int>(n));
    cout << "Nhap cac phan tu cua ma tran A:\n";
    for (int i = 0; i < m; ++i)
        for (int j = 0; j < n; ++j)
            cin >> A[i][j];

    int p;
    cout << "Nhap kich thuoc vector b: ";
    cin >> p;
    vector<int> b(p);
    cout << "Nhap cac phan tu cua vector b:\n";
    for (int i = 0; i < p; ++i)
        cin >> b[i];

    if (m != p) {
        cout << "Impossible\n";
    } else {
        cout << "Ma tran bo sung [A|b] la:\n";
        for (int i = 0; i < m; ++i) {
            for (int j = 0; j < n; ++j)
                cout << A[i][j] << " ";
            cout << b[i] << "\n";
        }
    }

    return 0;
}
